pages/front.js

```diff
@@ -1,31 +1,83 @@
+import { Component } from 'react'
 import { compose } from 'react-apollo'
+
+import { Router } from '../lib/routes'
 import Frame from '../components/Frame'
 import Front from '../components/Front'
+import StatusError from '../components/StatusError'
+
 import withData from '../lib/apollo/withData'
+import withMe from '../lib/apollo/withMe'
 import withMembership, { UnauthorizedPage } from '../components/Auth/withMembership'
-import StatusError from '../components/StatusError'
+import withInNativeApp from '../lib/withInNativeApp'
+
+import { Loader } from '@project-r/styleguide'
 
 const KNOWN_PATHS = ['/feuilleton']
 
-const FrontPage = (props) => {
-  if (props.isMember) {
-    return <Front {...props} />
+const isPathKnown = (url) => {
+  return KNOWN_PATHS.indexOf(url.asPath.split('?')[0]) !== -1
+}
+
+class FrontPage extends Component {
+  componentDidMount () {
+    this.redirectUser()
+  }
+
+  componentDidUpdate () {
+    this.redirectUser()
   }
-  const { url } = props
-  if (KNOWN_PATHS.indexOf(url.asPath.split('?')[0]) !== -1) {
-    return <UnauthorizedPage {...props} />
+
+  redirectUser () {
+    const { url, isMember } = this.props
+
+    // If path is known but user is not a member, redirect to index page.
+    if (isPathKnown(url) && !isMember) {
+      Router.pushRoute('index')
+    }
+  }
+
+  render () {
+    const { url, me, isMember, inNativeIOSApp, serverContext } = this.props
+
+    // If user is a member, render Front.
```

Comment:
I think the comment just repeats the code (which is good sign code-wise), so you could just get rid of the comment.